tinygrad/runtime/ops_amd.py

```diff
@@ -282,7 +283,9 @@ def __init__(self, device:AMDDevice, name:str, lib:bytes):
     # Some programs use hsa_kernel_dispatch_packet_t to read workgroup sizes during execution.
     # The packet is represented as a pointer and set up in SGPRs. Space for the packet is allocated as part of the kernel arguments.
     self.enable_dispatch_ptr = code.kernel_code_properties & hsa.AMD_KERNEL_CODE_PROPERTIES_ENABLE_SGPR_DISPATCH_PTR
+    self.enable_private_segment_sgpr = code.kernel_code_properties & hsa.AMD_KERNEL_CODE_PROPERTIES_ENABLE_SGPR_PRIVATE_SEGMENT_BUFFER
     additional_alloc_sz = ctypes.sizeof(hsa.hsa_kernel_dispatch_packet_t) if self.enable_dispatch_ptr else 0
+    if self.enable_private_segment_sgpr: additional_alloc_sz += 16
```

Comment:
This is not required. This alloc size of kernargs. `AMD_KERNEL_CODE_PROPERTIES_ENABLE_SGPR_DISPATCH_PTR` add a packet that is passed as a pointer, so it is embedded into kernargs. `AMD_KERNEL_CODE_PROPERTIES_ENABLE_SGPR_PRIVATE_SEGMENT_BUFFER` is just registers, so should be fine?